packages/types/src/extrinsic/Doughnut.ts

```diff
@@ -12,32 +12,32 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-import {AccountId, Signature, Struct, Text, Tuple, U32, U64, Vector} from '@plugnet/types';
+import {Bytes, Compact, U8a} from '@plugnet/types';
+import {AnyU8a} from '@plugnet/types/types';
 
-export class Certificate extends Struct {
-    constructor(value?: any) {
-        super(
-            {
-                expires: U64,
-                version: U32,
-                holder: AccountId,
-                notBefore: U64,
-                permissions: Vector.with(Tuple.with([Text, Text])),
-                issuer: AccountId,
-            },
-            value
-        );
+/**
+ * An encoded, signed v0 Doughnut certificate
+ **/
+export class Doughnut extends U8a {
+    get encodedLength(): number {
+        return this.toU8a().length;
+    }
+
+    constructor(value?: AnyU8a) {
```

Comment:
Can't find where doughnut is unboxed from compact when decoding.

Reply:
I've used `Bytes` which should handle the length prefix